azuredevops/internal/service/taskagent/resource_variable_group.go

```diff
@@ -375,6 +375,21 @@ func expandVariableGroupParameters(clients *client.AggregatedClient, d *schema.R
 	description := converter.String(d.Get(vgDescription).(string))
 	variables := d.Get(vgVariable).(*schema.Set).List()
 
+	// needed to detect if the secret_value attribute is set in the config
+	// see https://github.com/hashicorp/terraform-plugin-sdk/issues/741
+	for it := d.GetRawConfig().AsValueMap()[vgVariable].ElementIterator(); it.Next(); {
+		_, ctyVariable := it.Element()
+		ctyVariableAsMap := ctyVariable.AsValueMap()
+		name := ctyVariableAsMap[vgName].AsString()
+		valueSet := !ctyVariableAsMap[vgValue].IsNull()
+		secretValueSet := !ctyVariableAsMap[secretVgValue].IsNull()
+		isSecretSet := !ctyVariableAsMap[vgIsSecret].IsNull()
+
+		if valueSet && (secretValueSet || isSecretSet) || secretValueSet != isSecretSet {
+			return nil, nil, fmt.Errorf("`%s` variable can have either only `value` attribute or both `is_secret` and `secret_value` attributes", name)
```

Comment:
I think it might be clearer for the user if you split this up in 2 error cases/messages:
1. you can't set both `value` and `secret_value`
2. If `secret_value` is set, you must set `is_secret = true`

I think a combination of these two cases/messages will guide the user to the correct configuration for their variable. And the messages can be a bit clearer than now, trying to explain the relationship between the 3 in one sentence.